### DISCUSSION TITLE
How to distinguish between rename events and create events

### DISCUSSION THREAD

**Discussion by 27149chen** [seaweedfs/seaweedfs#6178] [Q&A]
According to the doc: https://github.com/seaweedfs/seaweedfs/wiki/Filer-Metadata-Events#what-kind-of-events-it-contains, rename events is different from create and update events, but in fact it is the same with a create event.

here is my rename event, i renamed /loutest/6 to /loutest/path1/6
```
{
  "directory": "/loutest/path1",
  "event_notification": {
    "new_entry": {
      "name": "6",
      "chunks": [
        {
          "size": 5,
          "modified_ts_ns": 1730257607325036000,
          "e_tag": "ZgtBYWfn/YObwGxhu1oYSw==",
          "fid": {
            "volume_id": 8,
            "file_key": 109,
            "cookie": 853869458
          },
          "is_compressed": true
        }
      ],
      "attributes": {
        "file_size": 5,
        "mtime": 1730257607,
        "file_mode": 420,
        "uid": 501,
        "gid": 20,
        "crtime": 1730257607,
        "mime": "text/plain; charset=utf-8",
        "inode": 17608011950027522301
      }
    },
    "delete_chunks": true,
    "new_parent_path": "/loutest/path1",
    "signatures": [
      1236864154,
      -979691841
    ]
  },
  "ts_ns": 1730257684284625000
}
```

**Comment by chrislusf** [comment]
rename would have `old_entry`

**Reply by 27149chen** [reply]
I can not find such events, just the one above

**Reply by 27149chen** [reply]
@chrislusf it's easy to reproduce, just start a mount and use mv inside it

**Reply by TsengSR** [reply]
@chrislusf 

I'm observing the same with S3 (WinSCP) and WebDAV (via Windows 11 Network Drive Mapping).

A rename always sends delete + create/added event. Is it different based on the way you access the storage?
Even so when moving the same file within the same bucket.

**Reply by TsengSR** [reply]
Also not sure the "update" is correct either. On Update/Modify I get old_entry and new_entry and a non-null new_parent_path, but [Docs](https://github.com/seaweedfs/seaweedfs/wiki/Filer-Metadata-Events#what-kind-of-events-it-contains) state `new_parent_path` is set on rename

**Comment by 27149chen** [comment]
tried again by running: `mv 3 4`, still no old entry

```
{
  "directory": "/tmp/loutest",
  "event_notification": {
    "new_entry": {
      "name": "4",
      "chunks": [
        {
          "size": 2,
          "modified_ts_ns": 1726116263107347000,
          "e_tag": "sCYyTGkEsqnLS4jW1hyB0Q==",
          "fid": {
            "volume_id": 8,
            "file_key": 94,
            "cookie": 1156223458
          },
          "is_compressed": true
        }
      ],
      "attributes": {
        "file_size": 2,
        "mtime": 1726116263,
        "file_mode": 420,
        "uid": 501,
        "gid": 20,
        "crtime": 1726116263,
        "mime": "text/plain; charset=utf-8",
        "inode": 14219030884291337544
      }
    },
    "delete_chunks": true,
    "new_parent_path": "/tmp/loutest",
    "signatures": [
      -1470161406,
      -979691841
    ]
  },
  "ts_ns": 1730348208701552000
}
```

**Comment by chrislusf** [comment]
This is fixed in https://github.com/seaweedfs/seaweedfs/pull/8854 . Renames now emit a single event with both `old_entry` and `new_entry` instead of separate create + delete events.
